cpp/lib/src/MarcRecord.cc

```diff
@@ -0,0 +1,296 @@
+#include "MarcRecord.h"
+#include "util.h"
+#include "MarcReader.h"
+
+std::string MarcRecord::getFieldData(const std::string &tag) const {
+    return getFieldData(getFieldIndex(tag));
+}
+
+
+std::string MarcRecord::getFieldData(const size_t index) const {
+    if (index == MarcRecord::FIELD_NOT_FOUND || directory_entries_.cbegin() + index >= directory_entries_.cend())
```

Comment:
Please use "or" instead of "||".  Also, why not replace this with

if (index > directory_entries_.size())  ?